### PR TITLE
KAFKA-15527: Update Javadoc for range and reverseRange in ReadOnlyKeyValueStore and update upgrade-guide.html for kafka streams webpage

### PR DESCRIPTION
**Pull request by hanyuzheng7**
Update Javadoc for range and reverseRange in ReadOnlyKeyValueStore.
Update upgrade-guide.html for kafka streams webpage.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -133,6 +133,15 @@ <h3 class="anchor-heading"><a id="streams_notable_changes" class="anchor-link"><
...
     </p>
 
+    <h3><a id="streams_api_changes_370" href="#streams_api_changes_370">Streams API changes in 3.7.0</a></h3>
+    <p>
+        IQv2 supports a <code>RangeQuery</code> that allows to specify unbounded, bounded, or half-open key-ranges.
+        Now, we add reverseRange and reverseAll query for <code>RangeQuery</code>. The concepts of reverseRange and reverseAll are not tied to any specific method or class.
+        Rather, they represent functionalities we wish to achieve. To achieve the functionalities of reverseRange and reverseAll, we can introduce a method named <code>withDescendingKeys()</code> for reversed queries.
+        Utilizing these, the query results are ordered based on the serialized byte[] of the keys, not the 'logical' key order.
+        <a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-985%3A+Add+reverseRange+and+reverseAll+query+over+kv-store+in+IQv2">KIP-985</a>
+    </p>
```
Comment (mjsax) [inline docs/streams/upgrade-guide.html:140]: I think we can simplify the paragraph (we try to be brief):
```
IQv2 supports <code>RangeQuery</code> that allows to specify unbounded, bounded, or half-open key-ranges, which return data in ascending (byte[]-lexicographical) order (per partition).
<a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-985%3A+Add+reverseRange+and+reverseAll+query+over+kv-store+in+IQv2">KIP-985</a> extends this functionality by adding <code>.withDescendingKeys()<code> to allow user to receive data in descending order.
```

File: streams/src/main/java/org/apache/kafka/streams/state/ReadOnlyKeyValueStore.java [hunk 2]
```diff
@@ -48,13 +48,13 @@ public interface ReadOnlyKeyValueStore<K, V> {
      * Get an iterator over a given range of keys. This iterator must be closed after use.
      * The returned iterator must be safe from {@link java.util.ConcurrentModificationException}s
      * and must not return null values.
-     * Order is not guaranteed as bytes lexicographical ordering might not represent key order.
+     * Order based on the serialized byte[] of the keys, not the 'logical' key order.
```
Comment (mjsax) [inline streams/src/main/java/org/apache/kafka/streams/state/ReadOnlyKeyValueStore.java]: `Order [is] based`

**Review by mjsax — COMMENTED** [review]
I think we should also improve `RangeQuery` JavaDocs in addition, to be clear about ordering guarantees there, too.

**Comment by hanyuzheng7** [comment]
> I think we should also improve `RangeQuery` JavaDocs in addition, to be clear about ordering guarantees there, too.

ok

File: streams/src/main/java/org/apache/kafka/streams/state/ReadOnlyKeyValueStore.java (hunk 2 above)
Comment (hanyuzheng7) [reply streams/src/main/java/org/apache/kafka/streams/state/ReadOnlyKeyValueStore.java]: ok, I will fix it.

File: docs/streams/upgrade-guide.html (hunk 1 above)
Comment (hanyuzheng7) [reply docs/streams/upgrade-guide.html:140]: ok, I will update it

**Pull request merged**